plugin/loadbalance/setup.go

```diff
@@ -10,34 +13,102 @@ import (
 )
 
 var log = clog.NewWithPlugin("loadbalance")
+var errOpen = errors.New("Weight file open error")
 
 func init() { plugin.Register("loadbalance", setup) }
 
 func setup(c *caddy.Controller) error {
-	err := parse(c)
+	policy, weighted, err := parse(c)
 	if err != nil {
 		return plugin.Error("loadbalance", err)
 	}
 
+	if policy == weightedRoundRobinPolicy {
```

Comment:
Can you move this logic into `parse`, and have `parse` return onStartup/onShutdown functions instead of `weightedRR` ?

Reply:
Done.